script.skinvariables/resources/lib/script.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Module: default
+# Author: jurialmunkey
+# License: GPL v.3 https://www.gnu.org/copyleft/gpl.html
+import sys
+import xbmc
+import xbmcgui
+import xbmcvfs
+from json import loads
+import resources.lib.utils as utils
+import xml.etree.ElementTree as ET
+
+
+class Script(object):
+    def __init__(self):
+        self.home = xbmcgui.Window(10000)
+        self.params = {}
+
+    def get_params(self):
+        for arg in sys.argv:
+            if arg == 'script.py':
+                pass
+            elif '=' in arg:
+                arg_split = arg.split('=', 1)
+                if arg_split[0] and arg_split[1]:
+                    key, value = arg_split
+                    self.params.setdefault(key, value)
+            else:
+                self.params.setdefault(arg, True)
+
+    def make_variables(self):
+        p_dialog = xbmcgui.DialogProgressBG()
+        p_dialog.create('Skin Variables', 'Constructing Variables...')
```

Comment:
non localized strings are not allowed in the ui

Reply:
I'm happy to localize. It just felt like overkill to localize something that no end user should ever see -- only skinners should be running this script.

Usage pattern is:
- Skinner runs the script to generate a `script-skinvariables-includes.xml` file
- Skinner commits the generated file to their repo

There is no reason for this script to run on an end-user machine. The generated file is the same for every user and the script doesn't do anything once the file has been generated.

I only made it an addon to simplify things a bit for skinners because it means you can add something like
`<onload condition="System.HasAddon(script.skinvariables)">Runscript(script.skinvariables)</onload>` to the Home.xml and then it will auto rebuild whenever the template file changes. Since the end-user has no reason to install the script, it will never run for them.

I'm not sure if only being for skinners and not end-users means that it maybe shouldn't be on the repo?

Reply:
ok all good